code/go/0chain.net/miner/protocol_round.go

```diff
@@ -299,13 +302,48 @@ func (mc *Chain) RedoVrfShare(ctx context.Context, r *Round) bool {
 	return false
 }
 
+func (mc *Chain) getClientState(
+	ctx context.Context,
+	roundNumber int64,
+) *util.MerklePatriciaTrie {
+	pround := mc.GetRound(roundNumber - 1)
+	pb := mc.GetBlockToExtend(ctx, pround)
+	pndb := pb.ClientState.GetNodeDB()
+	root := pb.ClientStateHash
+	mndb := util.NewMemoryNodeDB()
+	ndb := util.NewLevelNodeDB(mndb, pndb, false)
+	clientState := util.NewMerklePatriciaTrie(ndb, util.Sequence(roundNumber-1), root)
+	return clientState
+}
+
+func (mc *Chain) getConfigMap(clientState *util.MerklePatriciaTrie) (*minersc.GlobalSettings, error) {
```

Comment:
The `mc` is not used in `getConfigMap`, we don't need to put the function as a method of `mc`.